### PR TITLE
Rollback gem deprecate

### PR DESCRIPTION
**Pull request by bronzdoc**
# Description:
closes https://github.com/rubygems/rubygems/issues/3512

This PR does a few things:
- Restore old depreciation method 
- Deprecates old deprecation method
- Rename new version horizon deprecation methods

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by simi** [comment]
I think this is good enough for now. We can take a look at internal deprecation improvements later.

File: lib/rubygems/deprecate.rb [hunk 1]
```diff
@@ -45,13 +45,38 @@ def self.next_rubygems_major_version # :nodoc:
...
+      old = "_deprecated_#{name}"
+      alias_method old, name
+      define_method name do |*args, &block|
+        klass = self.kind_of? Module
+        target = klass ? "#{self}." : "#{self.class}#"
+        msg = [ "NOTE: #{target}#{name} is deprecated",
+                repl == :none ? " with no replacement" : "; use #{repl} instead",
+                ". It will be removed on or after %4d-%02d-01." % [year, month],
+                "\n#{target}#{name} called from #{Gem.location_of_caller.join(":")}",
+        ]
```
Comment (deivid-rodriguez) [inline lib/rubygems/deprecate.rb:65]: I believe we are introducing here the same kind of issue we're trying to solve in #3506: we're showing a deprecation message only meant for gem authors to all users. And we're showing this deprecation message about the deprecation system right next to the deprecation of a method that end users _do_ care about.

I feel we might cause brain damage to our users :laughing:.

So, I believe we should revert the deprecation of `Gem::Deprecate` for now until we figure out good strategies to be smart about whether we are in a end user context or in a gem development context, and apply them to this particular case.

Comment (simi) [reply lib/rubygems/deprecate.rb:65]: I have missed that. I agree on removing this deprecation warning for now.

Comment (deivid-rodriguez) [reply lib/rubygems/deprecate.rb:65]: FWIW, I placed this comment at the wrong place, my comments refers to the line _below_ this one, just to clarify.

File: lib/rubygems/deprecate.rb [hunk 2]
```diff
@@ -70,7 +95,7 @@ def deprecate(name, replacement=:none)
   end
 
   # Deprecation method to deprecate Rubygems commands
-  def deprecate_command
```
Comment (deivid-rodriguez) [inline lib/rubygems/deprecate.rb:73]: I think the names convey it quite well, but still we should make it super clear that this is an internal module, not a public one that can be used by anyone. Not sure which strategies are tipically used for this.

File: lib/rubygems/deprecate.rb [hunk 3]
```diff
@@ -45,13 +45,38 @@ def self.next_rubygems_major_version # :nodoc:
...
+  end
+
   ##
   # Simple deprecation method that deprecates +name+ by wrapping it up
   # in a dummy method. It warns on each call to the dummy method
   # telling the user of +repl+ (unless +repl+ is :none) and the
   # Rubygems version that it is planned to go away.
 
-  def deprecate(name, replacement=:none)
+  def rubygems_deprecate(name, replacement=:none)
```
Comment (deivid-rodriguez) [inline lib/rubygems/deprecate.rb:79]: I think the new name conveys it quite well, but we should make it super clear that this is an internal module, not a public one for everyone to use. Not sure which strategies are tipically used for this.

File: lib/rubygems/deprecate.rb (hunk 1 above)
Comment (bronzdoc) [reply lib/rubygems/deprecate.rb:65]: I'm not sure if there's other way to do this without end users seeing the deprecation warning. 🤔 

This is not something that we could check at build time...

Comment (deivid-rodriguez) [reply lib/rubygems/deprecate.rb:65]: I think we can try to approximate it. I first idea that came to my mind was whether there's a ".gemspec" file at the CWD or up in the directory hierarchy.

Comment (simi) [reply lib/rubygems/deprecate.rb:65]: Can we just remove it for now (should I provide PR?) and discuss this as a follow-up in new issue?

Comment (deivid-rodriguez) [reply lib/rubygems/deprecate.rb:65]: Yes, yes, that's exactly what I was suggesting.

Comment (simi) [reply lib/rubygems/deprecate.rb:65]: https://github.com/rubygems/rubygems/pull/3536

**Pull request merged**